src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.RegularCompilationTracker_Generators.cs

```diff
@@ -296,6 +300,9 @@ await newGeneratedDocuments.States.Values.SelectAsArrayAsync(
             var generationDateTime = DateTime.Now;
             foreach (var generatorResult in runResult.Results)
             {
+                if (generatorResult.GeneratedSources.IsDefault)
```

Comment:
wat?

must explain.

Reply:
I assume this means the generator was skipped, but it seems an explicit flag would be good. I'm also not sure what the _other_ members are in this case.

This also seems like it's going to break the optimization a few lines back with the "We'll first check the count of trees" comment. In the case of the razor generator being ran but not producing anything, we'll invalidate the compilation that might have had other generated files but -- by definition -- they didn't change.